mat/list_test.go

```diff
@@ -648,45 +769,86 @@ func underlyingData(a Matrix) []float64 {
 }
 
 // testMatrices is a list of matrix types to test.
-// The TriDense types have actual sizes because the return from Triangular is
-// only valid when n == 0.
+// This test relies on the fact that the implementations of Triangle do not
+// corrupt the value of Uplo when they are zero-valued. This test will fail
+// if that changes (and some mechanism will need to be used to force the
+// correct TriKind to be read).
 var testMatrices = []Matrix{
 	&Dense{},
-	&SymDense{},
-	NewTriDense(3, true, nil),
-	NewTriDense(3, false, nil),
-	&basicVector{},
-	&VecDense{mat: blas64.Vector{Inc: 1}},
-	&VecDense{mat: blas64.Vector{Inc: 10}},
-	&DiagDense{},
```

Comment:
Why does the zero value go away?

Reply:
It went away because it's redundant with having an incremented and a non-incremented `Dense` type.

Reply:
Okay, sorry, fixed an issue with the increments, and now the zero value is back (this is much better)